lhotse/dataset/speech_recognition.py

```diff
@@ -57,3 +58,122 @@ def __getitem__(self, idx: int) -> Dict[str, torch.Tensor]:
+                    'fsa': k2.Fsa,
+                    'start_frame': int,
+                    'end_frame': int
+                } (multiplied N times, for each of the N supervisions present in the Cut)
+            ],
+            'supervisions_mask': (T) tensor
+        }
+
+    The ``supervisions_mask`` field is a mask that specifies which frames are covered by a supervision
+    by assigning a value of 1 (in this case: segments with transcribed speech contents),
+    and which are not by asigning a value of 0 (in this case: padding, contextual noise,
+    or in general the acoustic context without transcription).
+    """
+
+    def __init__(self, cuts: CutSet):
+        import k2
+        super().__init__()
+        self.cuts = cuts
+        self.cut_ids = list(self.cuts.ids)
+
+        # We're creating a vocabulary of all characters present in the training CutSet.
+        # To do that we iterate through all the supervisions, and use chain.from_iterable
+        # to convert a sequence of their 'text' fields into a flat iterable of characters.
+
+        # Add special symbols first
+        self.vocab = ['<eps>', '<sil>', '<unk>'] + sorted(
+            # Add symbols present in the corpus
+            set(
+                chain.from_iterable(
+                    list(supervision.text)
+                    for cut in self.cuts
+                    for supervision in cut.supervisions
+                )
+            )
+        )
+        self.symbol_table = k2.SymbolTable(
+            _id2sym={i: tok for i, tok in enumerate(self.vocab)},
+            _sym2id={tok: i for i, tok in enumerate(self.vocab)}
+        )
+
+    def __getitem__(self, idx: int) -> Dict[str, torch.Tensor]:
+        import k2
+        cut_id = self.cut_ids[idx]
+        cut = self.cuts[cut_id]
+
+        features = torch.from_numpy(cut.load_features())
+        mask = torch.from_numpy(cut.supervisions_feature_mask())
+
+        sil_id = self.symbol_table.get('<sil>')
+        eps_id = self.symbol_table.get('<eps>')
+
+        supervision_fsas = []
+        for supervision in cut.supervisions:
+            # TODO(pzelasko): I am not sure how to attach the symbol table to the FSA object
+            #                 (or whether that's need at all?)
+
+            # Creating the core supervision FSA corresponding to the supervision text.
+            fsa = k2.linear_fsa([self.symbol_table.get(c) for c in supervision.text])
+
+            # TODO: Adding optional silence at beginning and end.
+
+            # TODO(pzelasko): Should be possible to create it with sth like:
+            #   silfsa = k2.union(
+            #     k2.linear_fsa([sil_id]), k2.linear_fsa([eps_id])
+            #   ).closure()
+            silfsa = k2.Fsa.from_str(f'''0 1 {sil_id} 0
+                                         0 1 {eps_id} 0
+                                         1 0 {sil_id} 0
+                                         1''')
+            # TODO(pzelasko): is it possible to do:
```

Comment:
@danpovey 
Do we need to implement `concat` in k2?

Reply:
We'll definitely need `concat` at some point.  
If we use the L.fst that comes from kaldi, I don't think  we'd need the `silfsa` because L.fst provides the optional silence itself.

Reply:
In that case, do you want to use phones to create the supervision FSAs from utterances (i.e. create the word-level symbol table and compose with L.fst before returning from __getitem__)?